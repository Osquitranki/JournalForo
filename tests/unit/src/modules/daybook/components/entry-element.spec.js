import { shallowMount } from "@vue/test-utils";
import EntryElement from "@/modules/daybook/components/Entry-element.vue"
import {journalState} from "../../../mock-data/test-journal-state"


describe('pruebas en Entry component', () => {

const mockRouter = {
        push : jest.fn()
}


const wrapper = shallowMount(EntryElement , {
        props: {
            entry :  journalState.entries[0]
        },
        global: {
            mocks: {
                $router: mockRouter
             }
 
        }
})


/*  test('debe de hacer match con el snapshop', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  test('debe de redireccionar al hacer clik en el entry container', async() => {
    const entryContainer = wrapper.find('.entry-container')
    entryContainer.trigger('click')
    expect(mockRouter.push).toHaveBeenCalled()
    expect(mockRouter.push).toHaveBeenCalledWith(
        {name: 'entry',
         params: {id:journalState.entries[0].id}
        }
    )
    expect(mockRouter.push).toHaveBeenCalledWith(
        {name: 'entry',
         params: {id:expect.any(String)}
        }
    )
  })*/

  test('propiedades computadas', () => {
    console.log('VM', wrapper.vm.day)
    console.log('VM', wrapper.vm.month)
    console.log('VM', wrapper.vm.yearDay)
    expect(wrapper.vm.day).toBe(17)
    expect(wrapper.vm.month).toBe('Enero')
    expect(wrapper.vm.yearDay).toBe('2023, Martes')
  })
})
