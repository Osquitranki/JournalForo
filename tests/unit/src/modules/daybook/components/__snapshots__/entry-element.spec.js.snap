// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en Entry component debe de hacer match con el snapshop 1`] = `
<div class="entry-container mb-3 pointer p-2">
  <div class="entry-title d-flex"><span class="text-sucess fs-5 fw-blod">17</span><span class="mx-1 fs-5">Enero</span><span class="mx-2 fw-light">2023, Martes</span></div>
  <div class="entry-description">Elefantemod</div>
</div>
`;
